pkg/cluster/sync.go

```diff
@@ -309,7 +317,7 @@ func (c *Cluster) syncPodDisruptionBudget(isUpdate bool) error {
 	return nil
 }
 
-func (c *Cluster) syncStatefulSet() error {
+func (c *Cluster) syncStatefulSet(force bool) error {
```

Comment:
is there way we move all checks done outside to set the value of `force` inside syncStatefulSet? It's called in so many places and unit test and I find the extra argument kinda ugly.


In general, I like that we call syncStatefulSet without extra conditions and resolve it inside the function.